Add manifests for 0.12.0

skitt opened the pull request:
Fixes: #1716
Signed-off-by: Stephen Kitt <skitt@redhat.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

skitt committed:
Add manifests for 0.12.0

Fixes: #1716
Signed-off-by: Stephen Kitt <skitt@redhat.com>

skitt committed:
Test deployments using all operator-defined tools

This ensures that changes made here don't break subctl, bundle and
OCM-based deployments.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

skitt commented:
This gets us past the bundle validation errors seen in the periodic check, but the deployment itself fails.

mkolesnik commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,57 @@
+---
```
Why are you adding this entire job matrix? It seems quite excessive and most of it (if not all) should be covered by the E2E

skitt replied:
It’s covered in Shipyard, but not here. There are _no_ tests of the bundle in the operator, which is unfortunate since that’s where it gets changed. Catching failures in Shipyard doesn’t prevent merging the PR causing them in the operator.

mkolesnik commented on config/manifests/kustomization.yaml:
```diff
@@ -9,6 +9,6 @@ resources:
 - name: controller
   newName: quay.io/submariner/submariner-operator
-  newTag: 0.11.0
+  newTag: devel-efc6ef9e
```
Is it correct to reference devel?

skitt replied:
No, and I think this is contributing to the failures of the bundle test jobs; the published bundles now store the hash of the operator repository, rather than the relevant tag.

Commit:
Update base image to use stable branch 'release-0.12'

Signed-off-by: Automated Release <release@submariner.io>

skitt committed:
Bump Submariner dependencies and pin Shipyard rc0

This ensures we have a consistent set of dependencies (notably,
Lighthouse e2e with regards to the Shipyard e2e framework changes)
for 0.12.0-rc0.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

submariner-bot committed:
Pin Admiral to v0.12.0-rc0 (#1889)

Signed-off-by: Automated Release <release@submariner.io>

Co-authored-by: Automated Release <release@submariner.io>

submariner-bot committed:
Update Operator to use version v0.12.0-rc0 (#1891)

Signed-off-by: Automated Release <release@submariner.io>

Co-authored-by: Automated Release <release@submariner.io>

dfarrell07 committed:
Add packagedoc-lint linting, fix errors

The idea here is to make sure the license headers are detached from the
package declarations, as otherwise the headers are considered a part of
the package documentation and included with it.

Add per-PR job to maintain consistent standards across all repos.

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>
(cherry picked from commit 01a78a5af01d0d01b3aa63d228f617367a5849f0)

tpantelis committed:
Fix extraneous uninstall pods present after cleanup complete

After the finalizer is finalizer, it can take a little time before K8s
deletes the object. In the mean time, the controller may get triggered
again in which case it will observe the DeletionTimeStamp non-zero and
re-create the uninstall daemon sets. To avoid this, elide running
cleanup if the finalizer isn't present.

Also increased the requeue interval to 500 ms.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>
(cherry picked from commit a1dbdeebe3c7f39a14784d353c81569307dc740f)

skitt committed:
Always end gather status

Currently, info.Status.End() is only called if the module supports the
requested datatype. This means that for datatypes which aren't
supported, the spinner started by info.Status.Start() continues
running until the program exits.

Fixes: #1904
Signed-off-by: Stephen Kitt <skitt@redhat.com>
(cherry picked from commit c6429094583fe62a5e88329f1cbc7a6436d99d55)

mkolesnik committed:
Change uninstall to be reentrant

This will be better UX and make sure that users can still uninstall even
if for some reason a previous uninstall didn't finish successfully.

Resolves: #1903

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>
(cherry picked from commit 821456f66dbcfe9842f5e791bd9f67cedf047cbc)

mkolesnik committed:
Add cluster name to `subctl uninstall`

Resolves: #1897

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>
(cherry picked from commit c9090af)

yboaron committed:
Fix subctl show networks and versions output

Signed-off-by: yboaron <yboaron@redhat.com>
(cherry picked from commit e644897b3dbc0fa00733ef324b512ca1b3acb250)

mkolesnik committed:
Capture API warnings on `subctl gather`

This will make sure any warnings, such as deprecation, are properly
captured and reported so that they can be later handled.

Resolves: #1896

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>
(cherry picked from commit 6c3a1c7f390135ff1aac0eca435a3e59dd3ee40e)

mkolesnik committed:
Print out brokers information on `subctl show all`

Print out the info, even if clusters only have brokers.
Clusters that don't have brokers will just show informatory "Detecting
brokers" message and nothing further.

Resolves: #1894

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>
(cherry picked from commit 56dbe5102c3c0cddabb4ef951ddcd46e9cb9f047)

sridhargaddam committed:
Modify default state of dedicated-gw for GCP

Fixes: https://github.com/submariner-io/submariner-operator/issues/1915
Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>
(cherry picked from commit 664e99e5f712bc5ffca62f109027947620d633c2)

sridhargaddam committed:
Modify default state of dedicated-gw for rhos

Currently the default value of dedicated-gw is set as false, but
this could potentially have issues during cluster scaling and/or
during upgrades. The recommended setting is to use dedicated gw
nodes. This PR modifies the default value accordingly.

Fixes: https://github.com/submariner-io/submariner-operator/issues/1919
Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>
(cherry picked from commit 8a27658c9ddd8e196ec2db00b5645818ade1b3d5)

mkolesnik committed:
Reduce verbosity of controller reconciliation

These messages are flooding the log and making it grow endlessly, while
hiding real issues, as the loop happens every few seconds.
Reducing the verbosity of info logs helps clear the log and keep it
tidy.

Resolves: #1914

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>
(cherry picked from commit d1b458a9e5fd4f9b1708d40bbe1ff4cbf69543e9)

submariner-bot committed:
Pin Shipyard to v0.12.0-rc1 (#1935)

Signed-off-by: Automated Release <release@submariner.io>

Co-authored-by: Automated Release <release@submariner.io>

submariner-bot committed:
Pin Admiral to v0.12.0-rc1 (#1938)

Signed-off-by: Automated Release <release@submariner.io>

Co-authored-by: Automated Release <release@submariner.io>

submariner-bot committed:
Update Operator to use version v0.12.0-rc1 (#1939)

Signed-off-by: Automated Release <release@submariner.io>

Co-authored-by: Automated Release <release@submariner.io>

skitt committed:
Use the Kubernetes RFC-1123 validation

This simplifies our code. Add tests while we're at it.

Signed-off-by: Stephen Kitt <skitt@redhat.com>
(cherry picked from commit 7a521c6c3087a6bdc660c23820b00b3dca323343)

Jaanki committed:
Revert "Use the Kubernetes RFC-1123 validation"

This reverts commit fd135b7395aa2d3da0099cca9dd3f0e926632510 as it was
untimely merged and prevents from proceeding with the GA release after
the latest RC1 release. This change will be reintroduced once the GA
is released.

Signed-off-by: Janki Chhatbar <jchhatba@redhat.com>

submariner-bot committed:
Pin Shipyard to v0.12.0 (#1966)

Signed-off-by: Automated Release <release@submariner.io>

Co-authored-by: Automated Release <release@submariner.io>

submariner-bot committed:
Pin Admiral to v0.12.0 (#1967)

Signed-off-by: Automated Release <release@submariner.io>

Co-authored-by: Automated Release <release@submariner.io>

submariner-bot committed:
Update Operator to use version v0.12.0 (#1969)

Signed-off-by: Automated Release <release@submariner.io>

Co-authored-by: Automated Release <release@submariner.io>

dfarrell07 commented:
This depends on other bundle work, closing for now.
